core/core-impl/src/main/java/ai/timefold/solver/core/config/solver/EnvironmentMode.java

```diff
@@ -19,6 +19,26 @@
  */
 @XmlEnum
 public enum EnvironmentMode {
+    /**
+     * This mode turns on variable tracking and all assertions to fail-fast on a bug in a {@link Move} implementation, a
+     * constraint, the engine itself or something else at an appalling performance cost.
+     * <p>
+     * Because it tracks genuine and shadow variables, it is able to report precisely what variables caused the corruption and
+     * report any missed {@link ai.timefold.solver.core.api.domain.variable.VariableListener} events.
+     * <p>
+     * When undo score corruption is detected, it throws an
+     * {@link ai.timefold.solver.core.api.solver.exception.UndoScoreCorruptionException}, which can be used to inspect the
+     * working solution state before, after and after the undo of a {@link Move}.
+     * <p>
+     * This mode is reproducible (see {@link #REPRODUCIBLE} mode).
+     * <p>
+     * This mode is intrusive because it calls the {@link InnerScoreDirector#calculateScore()} more frequently than a non assert
+     * mode.
+     * <p>
+     * This mode is appallingly slow.
+     */
+    FULL_ASSERT_WITH_TRACKING(true),
+
     /**
      * This mode turns on all assertions
```

Comment:
This no longer turns on _all_ assertions.

Reply:
It does turn on all assertions; `TRACKED_FULL_ASSERT` will only fail when `FULL_ASSERT` fails, the main difference is it tells you why it failed.